src/keys.test.ts

```diff
@@ -111,6 +112,25 @@ describe("typing", () => {
       expectTypeOf(result).toEqualTypeOf<[]>();
     });
 
+    test("unions of records", () => {
+      const dataFirst = keys(
+        {} as Record<number, string> | Record<string, string>,
+      );
+
+      expectTypeOf(dataFirst).toEqualTypeOf<
+        Array<`${number}`> | Array<string>
+      >();
+
+      const dataLast = pipe(
+        {} as Record<number, string> | Record<string, string>,
```

Comment:
You can pull out this as a data variable and reuse it in the 2 tests instead of repeating the type. That way, if someone changes the type for one test, they don't drift accidentally.

```suggestion
        data,
```

Reply:
I agree it would be better to extract a common `data` variable. I will integrate this change to the rest of the tests too, as you suggested.